src/amo/components/Page/index.js

```diff
@@ -72,8 +76,15 @@ export const PageBase = ({
     }
   }
 
+  const pageAddonType = addonType || location.query.addonType;
+
   return (
     <div className="Page-amo">
+      {pageAddonType === ADDON_TYPE_EXTENSION &&
```

Comment:
I think `Page` should have a prop named `showVPNBanner` (default would be `false`) and the `Addon` would be responsible for computing the value because we want to only show the banner on add-on detail pages (nowhere else AFAIK), for Desktop and for extension detail pages only.

Reply:
Fair enough, but how is the `Addon` page going to compute that value when it is in a loading state? That is the challenge here. 

Do you think we should develop a function that has the ability to search through every relevant state (e.g., `addonsByAuthors`, `autocomplete`, `home`, `landing`, `languageTools`, `recommended`, `search`, etc.) to find the add-on we're looking for and determine its type? Or should we have a separate state where we load every search result, regardless of its source, so we have one place to look? Or do you have another idea for how we can determine the `addonType` when the add-on is loading?

Reply:
sorry, I left a comment here to keep the discussion in #10773 focused on the problem with the banner to display only for extension detail pages.

Reply:
Yes, I see we're discussing this in two different places, which isn't ideal. I'll move my comments there as well.